internal/ir/ir.go

```diff
@@ -352,6 +352,20 @@ type HashMap struct {
 	Value Value
 }
 
+// SubscriptExpr is an IR Value that represents the access of an element of an array.
+//
+// Example printed form:
+// %t0 = array["a","b","c"]
+// %t1 = "0"
+// %t2 = %t0[%t1]
+//
+// The SubscriptExpr implements the Value interface.
+type SubscriptExpr struct {
+	node
+	Object Value // identifier of the array object
```

Comment:
I think that a better doc to this should be: `The object being subscripted by the index`